src/DispatchPlot.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+  Author: dylanjm
+  Date: 2021-05-18
+"""
+import os
+import sys
+import itertools as it
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from PluginBaseClasses.OutStreamPlotPlugin import PlotPlugin, InputTypes, InputData
+
+# Matplotlib Global Settings
+plt.rc("figure", figsize=(12, 8), titleweight='bold')
+plt.rc(
+  "axes",
+  titleweight="bold",
+  labelsize=12,
+  axisbelow=True,
+  grid=True
+)
+plt.rc("savefig", bbox="tight")
+plt.rc("legend", fontsize=12)
+plt.rc(["xtick", "ytick"], labelsize=10)
+
+
+class DispatchPlot(PlotPlugin):
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Define the acceptable user inputs for this class.
+      @ In, None
+      @ Out, specs, InputData.ParameterInput,
+    """
+    specs = super().getInputSpecification()
+    specs.addSub(InputData.parameterInputFactory('source', contentType=InputTypes.StringType))
+    return specs
+
+  def __init__(self):
+    """
+      Constructor.
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'HERON.DispatchPlot'
+    self._sourceName = None
+    self._source = None
+
+  def handleInput(self, spec):
+    """
+      Reads in data from the input file
+      @ In, spec, InputData.ParameterInput, input information
+      @ Out, None
+    """
+    super().handleInput(spec)
+    for node in spec.subparts:
+      if node.getName() == 'source':
+        self._sourceName = node.value
+
+  def initialize(self, stepEntities):
+    """
+      Set up plotter for each run
+      @ In, stepEntities, dict, entities from the Step
+      @ Out, None
+    """
+    super().initialize(stepEntities)
+    src = self.findSource(self._sourceName, stepEntities)
+    if src is None:
+      self.raiseAnError(IOError, f'Source DataObject {self._sourceName} was not found in the Step!')
```

Comment:
can we throw quotes around the name of the source data object? It's surprising how unclear these messages can appear without them for some users.

Reply:
Sure thing